### DISCUSSION TITLE
Smart reminders and alarms

### DISCUSSION THREAD

**Discussion by jodli** [NightscoutFoundation/xDrip#3633] [Ideas]
Hey!

After watching a webinar by Medtronic last week about their new [InPen smart insulin pen system](https://www.medtronicdiabetes.com/products/inpen-smart-insulin-pen-system) it had lots of similarities to how we use xDrip at the moment.

In short:
1. We use a CGM to get data into xDrip
1. We calculate carbs and add them to xDrip
1. We use the "Predictive Simulations" feature in xDrip to calculate the required Bolus (according to the configured ratios, sensitivity, absorption rate and "Insulin on Board" values)
1. We add the given Bolus to xDrip to complete the cycle

---

Now I saw a few general things in the presentation that I liked:
- "Check BG 2hrs after dose" that reminds you to check back after every given Bolus
- "Check BG at Bedtime" that reminds you to measure your BG before going to sleep
- "Long-acting Reminder" that reminds you to inject your Basal insulin
- "Dose reminder" that helps you avoid missed Bolus after logging carbs, but no dose

They also had some alarm features that according to them lead to less false positive alarms:
- High alarms that consider trends of the CGM and the IoB before alerting. See [here](https://resources.cloud.medtronic-diabetes.com/sites/prd/files/documents/2024-01/emea-cgm-2300109_smart_mdi_2.0_hcp_mgdl_0.pdf)
- Low alarms that consider trends of the CGM to avoid false positive alarms, e.g. compression lows

Is there anything in this list that's worth implementing in xDrip?

**Comment by Navid200** [comment]
xDrip alerts already take into account the trend.  

You don't say anything about persistent high alert.
Unfortunately, many have no idea what it is: https://navid200.github.io/xDrip/docs/Alerts/PersistentHigh.html  

We also have a forecast low alert: https://navid200.github.io/xDrip/docs/Alerts/ForecastLow.html  

xDrip has a reminders utility.  Have you seen it?

**Reply by jodli** [reply]
Yes, I know about the persistent high alert. It's way more useful to use that instead of the basic high alert. But nevertheless it does not take the insulin bolus dose into account, so it alerts even if I've already corrected with a dose.

I did not know about the reminder utility though. That's quite nice, thanks!
